internal/app/machined/internal/phase/rootfs/etc/etc.go

```diff
@@ -136,33 +73,28 @@ func OSRelease() (err error) {
 
 	err = tmpl.Execute(writer, data)
 	if err != nil {
-		return
-	}
-
-	if err = ioutil.WriteFile("/run/system/etc/os-release", writer.Bytes(), 0644); err != nil {
-		return fmt.Errorf("write /run/system/etc/os-release: %w", err)
-	}
-
-	if err = unix.Mount("/run/system/etc/os-release", "/etc/os-release", "", unix.MS_BIND, ""); err != nil {
-		return fmt.Errorf("failed to create bind mount for /etc/os-release: %w", err)
+		return err
 	}
 
-	return nil
+	return ioutil.WriteFile("/run/system/etc/os-release", writer.Bytes(), 0644)
 }
 
-func ip() string {
-	addrs, err := net.InterfaceAddrs()
-	if err != nil {
-		return ""
+// createBindMount creates a common way to create a writable source file with a
+// bind mounted destination. This is most commonly used for well known files
+// under /etc that need to be adjusted during startup.
+func createBindMount(src, dst string) (err error) {
+	var f *os.File
+
+	if f, err = os.OpenFile(src, os.O_WRONLY|os.O_CREATE, 0644); err != nil {
+		return err
 	}
 
-	for _, address := range addrs {
-		if ipnet, ok := address.(*net.IPNet); ok && !ipnet.IP.IsLoopback() {
-			if ipnet.IP.To4() != nil {
-				return ipnet.IP.String()
-			}
-		}
+	// nolint: errcheck
+	defer f.Close()
```

Comment:
I believe we don't really need to `defer` here, as we can close immediately?
```suggestion
	if err = f.Close(); err != nil {
	  return
	}
```